Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -341,9 +402,9 @@ void Connection::setShouldExitOnSyncMessageSendFailure(bool shouldExitOnSyncMess
 // IPC messages as some messages may get received on the IPC thread before the message receiver registered itself on the main thread.
 void Connection::enqueueMatchingMessagesToMessageReceiveQueue(MessageReceiveQueue& receiveQueue, const ReceiverMatcher& receiverMatcher)
 {
-    ASSERT(isMainRunLoop());
+    ASSERT_MAIN_RUN_LOOP_AND_SYNC_STATE;
```

Comment:
Let's not add convoluted macro-based solutions if possible.
This should be
`assertIsCurrent(dispatcher());`

Reply:
I got rid of the `dispatcher()` function, since it'd have to be `WTF_REQUIRES_LOCK` (since the dispatcher pointer can go away at any time, if you're not on the dispatcher thread).

That means this would have to be:

`{
    Locker lock { m_syncStateLock; }
    assertIsCurrent(dispatcher());
}`

For each of the places we check it. I realise that this is just an assert, and the dispatcher can't racily go away (assuming that the code is correct, and this assert always succeeds), but I don't think TSA will (or should) let us ignore that.